src/v2/api/services/DisciplineTeacherService.spec.ts

```diff
@@ -0,0 +1,207 @@
+import { Test } from '@nestjs/testing';
+import { DisciplineTeacherService } from './DisciplineTeacherService';
+import { DisciplineRepository } from '../../database/repositories/DisciplineRepository';
+import { TeacherRole } from '@prisma/client';
+import { PollService } from './PollService';
+import { DateModule } from '../../utils/date/DateModule';
+import { PrismaModule } from '../../modules/PrismaModule';
+import { TelegramAPI } from '../../telegram/TelegramAPI';
+import { MapperModule } from '../../modules/MapperModule';
+import { PollModule } from '../../modules/PollModule';
+import { ConfigurationModule } from '../../modules/ConfigModule';
+import Configuration from '../../config/Configuration';
+import { TelegramConfigService } from '../../config/TelegramConfigService';
+import { NotEnoughAnswersException } from '../../utils/exceptions/NotEnoughAnswersException';
+import { ExcessiveAnswerException } from '../../utils/exceptions/ExcessiveAnswerException';
+import { WrongTimeException } from '../../utils/exceptions/WrongTimeException';
+
+describe('DisciplineTeacherService', () => {
+  let disciplineTeacherService: DisciplineTeacherService;
+  let disciplineRepository: DisciplineRepository;
+  let pollService: PollService;
+  beforeEach(async () => {
+    const testingModule = await Test.createTestingModule({
+      providers: [DisciplineTeacherService, TelegramAPI, TelegramConfigService],
+      imports: [
+        PrismaModule,
+        PollModule,
+        DateModule,
+        MapperModule,
+        ConfigurationModule.forRoot({
+          isGlobal: true,
+          load: [Configuration],
+        })],
+    }).compile();
+
+    disciplineTeacherService = testingModule.get(DisciplineTeacherService);
+    disciplineRepository = testingModule.get(DisciplineRepository);
+    pollService = testingModule.get(PollService);
+  });
+
+  describe('getUniqueQuestions', () => {
+    it('should return array of question', async () => {
+      jest.spyOn(disciplineRepository, 'find').mockImplementation(async () => ({
+        disciplineTeachers: [{
+          id: 'id',
+          roles: [TeacherRole.LECTURER, TeacherRole.PRACTICIAN],
+        }],
+      } as any));
+      jest.spyOn(pollService, 'getQuestions').mockImplementation(() => ([
+        {
+          id: 'question1',
+        },
+        {
+          id: 'question2',
+        }] as any
+      ));
+			
+      const result = await disciplineTeacherService.getUniqueQuestions('id');
+      expect(result).toStrictEqual([
+        {
+          id: 'question1',
+        },
+        {
+          id: 'question2',
+        }]
+      );
+    });
```

Comment:
Here you are actually testing the actual mock, so I don't see the point of this test